lambdas/journeyengine/src/main/java/uk/gov/di/ipv/core/journeyengine/JourneyEngineHandler.java

```diff
@@ -142,34 +146,57 @@ private JourneyEngineResult executeJourneyEvent(
                             new JourneyResponse(criStartUri + UK_PASSPORT_CRI_ID));
                     break;
                 case CRI_UK_PASSPORT:
-                    updateUserState(CRI_ADDRESS, ipvSessionItem);
-                    builder.setJourneyResponse(new JourneyResponse(criStartUri + ADDRESS_CRI_ID));
+                    if (journeyStep.equals(NEXT_STEP)) {
+                        updateUserState(CRI_ADDRESS, ipvSessionItem);
+                        builder.setJourneyResponse(
+                                new JourneyResponse(criStartUri + ADDRESS_CRI_ID));
+                    } else {
+                        updateUserState(CRI_ERROR, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    }
                     break;
                 case CRI_ADDRESS:
-                    updateUserState(CRI_FRAUD, ipvSessionItem);
-                    builder.setJourneyResponse(new JourneyResponse(criStartUri + FRAUD_CRI_ID));
+                    if (journeyStep.equals(NEXT_STEP)) {
+                        updateUserState(CRI_FRAUD, ipvSessionItem);
+                        builder.setJourneyResponse(new JourneyResponse(criStartUri + FRAUD_CRI_ID));
+                    } else {
```

Comment:
I feel like this should still be a `else if` block, to make it clear that we're only doing this if the step is an error step. That way if we get an unexpected step for some reason it'll still drop out to the new `default` case.

Reply:
It won't go to the default case for this kind of scenario. That's only for the actual current user status, this is just checking which journey step we get

Reply:
Just to capture what we just chatted about IRL - we're going to make the `else` block an `else if` to be specific about the step we've received, and add an `else` block that will throw if we've received and unexpected step.